app-connector-client/server/routes/apis.js

```diff
@@ -39,49 +37,49 @@ function getAll(req, res) {
     }
 };
 
-function getAllAPIs(localKyma, cb) {
+function getAllAPIs(cb) {
     request({
-        url: CONFIG.URLs.metadataUrl,
+        url: connection.info().metadataUrl,
         method: "GET",
         agentOptions: {
-            cert: fs.readFileSync(certFile),
-            key: fs.readFileSync(keyFile)
+            cert: fs.readFileSync(connection.info().certificate),
+            key: fs.readFileSync(connection.privateKey())
         },
-        rejectUnauthorized: !localKyma
+        rejectUnauthorized: !connection.info().insecure
     }, function (error, httpResponse, body) {
         cb(error, httpResponse, body);
     });
 };
 
-function createAPI(localKyma, serviceMetadata, cb) {
+function createAPI(serviceMetadata, cb) {
     request.post({
-        url: CONFIG.URLs.metadataUrl,
+        url: connection.info().metadataUrl,
         headers: {
             "Content-Type": "application/json"
         },
         json: serviceMetadata,
         agentOptions: {
-            cert: fs.readFileSync(certFile),
-            key: fs.readFileSync(keyFile)
+            cert: fs.readFileSync(connection.info().certificate),
+            key: fs.readFileSync(connection.privateKey())
         },
-        rejectUnauthorized: !localKyma
+        rejectUnauthorized: !connection.info().insecure

```

Comment:
Also for this. I would prefer no direct access to objects returned by commands, but accessing them through setter. So `connection.isSecure()`

Reply:
Good point, I changed it to connection.secure() which makes usage much simpler. I think getters are not typical. I removed the "is" from "isEstablished" as well to have it consistent